File: Polimorfismo/ejercicio 3/ej3.java
class Empleado {
    String nombre;
    int sueldoMes;
    int horasExtra;
    float sueldoHora;
    float propina;
    String cargo;

    public Empleado(String nombre, int sueldoMes, int horasExtra, float sueldoHora, float propina, String cargo) {
        this.nombre = nombre;
        this.sueldoMes = sueldoMes;
        this.horasExtra = horasExtra;
        this.sueldoHora = sueldoHora;
        this.propina = propina;
        this.cargo = cargo;
    }

    public float sueldoTotal() {
        return sueldoMes + (horasExtra * sueldoHora) + propina;
    }

    public void mostrar() {
        System.out.println(cargo + ": " + nombre + ", Sueldo Total: Bs" + sueldoTotal());
    }

    public static void mostrarEmpleados(int sueldoMes, Empleado[] empleados) {
        System.out.println("\nEmpleados con sueldo mensual de " + sueldoMes + ":");
        for (Empleado empleado : empleados) {
            if (empleado.sueldoMes == sueldoMes) {
                System.out.println("- " + empleado.nombre + " (" + empleado.cargo + ")");
            }
        }
    }
}

public class Restaurante {
    public static void main(String[] args) {

        Empleado cocinero = new Empleado("Felipe", 2500, 10, 15, 0, "Cocinero");
        Empleado mesero1 = new Empleado("Marco", 1500, 5, 10, 200, "Mesero");
        Empleado mesero2 = new Empleado("Maria", 1500, 8, 10, 250, "Mesero");
        Empleado admin1 = new Empleado("Britney", 3400, 0, 0, 0, "Administrativo");
        Empleado admin2 = new Empleado("Alan", 3400, 5, 20, 0, "Administrativo");

        Empleado[] empleados = {cocinero, mesero1, mesero2, admin1, admin2};

        for (Empleado empleado : empleados) {
            empleado.mostrar();
        }

        Empleado.mostrarEmpleados(1500, empleados);
    }
}

File: Polimorfismo/ejercicio 3/ej3.py
class Empleado:
    def __init__(self, nombre, sueldo_mes, horas_extra=0, sueldo_hora=0, propina=0, cargo=""):
        self.nombre = nombre
        self.sueldo_mes = sueldo_mes
        self.horas_extra = horas_extra
        self.sueldo_hora = sueldo_hora
        self.propina = propina
        self.cargo = cargo

    def sueldo_total(self):
        return self.sueldo_mes + (self.horas_extra * self.sueldo_hora) + self.propina

    def mostrar(self):
        print(f"{self.cargo}: {self.nombre}, Sueldo Total: Bs{self.sueldo_total()}")

def mostrar_empleados(sueldo_mes, empleados):
    print(f"\nEmpleados con sueldo mensual de {sueldo_mes}:")
    for empleado in empleados:
        if empleado.sueldo_mes == sueldo_mes:
            print(f"- {empleado.nombre} ({empleado.cargo})")

cocinero = Empleado("Felipe", 2500, 10, 15, 0, "Cocinero")
mesero1 = Empleado("Marco", 1500, 5, 10, 200, "Mesero")
mesero2 = Empleado("Maria", 1500, 8, 10, 250, "Mesero")
admin1 = Empleado("Britney", 3400, 0, 0, 0, "Administrativo")
admin2 = Empleado("Alan", 3400, 5, 20, 0, "Administrativo")

empleados = [cocinero, mesero1, mesero2, admin1, admin2]

for empleado in empleados:
    empleado.mostrar()

mostrar_empleados(1500, empleados)